File: src/backend/Seems.Domain/Entities/AuditLog.cs
using Seems.Domain.Enums;

namespace Seems.Domain.Entities;

public class AuditLog
{
    public long Id { get; set; }
    public string EntityName { get; set; } = string.Empty;
    public string EntityId { get; set; } = string.Empty;
    public AuditAction Action { get; set; }
    public string? UserId { get; set; }
    public string? UserEmail { get; set; }
    public string? ChangedFields { get; set; }
    public DateTime Timestamp { get; set; }
}

File: src/frontend/admin/src/types/auditLog.ts
export type AuditAction = 'Created' | 'Updated' | 'Deleted'

export interface AuditLog {
  id: number
  entityName: string
  entityId: string
  action: AuditAction
  userId: string | null
  userEmail: string | null
  changedFields: string[] | null
  timestamp: string
}

export interface PaginatedAuditLogs {
  items: AuditLog[]
  total: number
  page: number
  pageSize: number
  totalPages: number
}

export interface AuditLogFilters {
  entityName: string | null
  action: string | null
  userEmail: string | null
  dateFrom: Date | null
  dateTo: Date | null
}
